src/components/layout/shared/ConnectButton.tsx

```diff
@@ -26,20 +29,17 @@ export const CustomConnectButton = () => {
     }
   }, [switchChain])
 
-  useEffect(() => {
-    const checkAndLinkWallet = async () => {
-      if (!github || isAdmin) return
-      const linkedAddress = await getLinkedWallet(github)
-      if (linkedAddress == '0x0000000000000000000000000000000000000000') {
-        if (!address) return
-        await linkWallet({
-          github,
-          address,
-        })
-      }
+  useAsyncEffect(async () => {
+    const linkedAddress = walletState.linkedAddress
+    if (!github || isAdmin || linkedAddress === '') return
+    if (linkedAddress == '0x0000000000000000000000000000000000000000') {
```

Comment:
how about use a constant file to store data like this. I guess there is some file named as data.ts? It's a mess, maybe add another commit to make that part better. And fix more in sperate pr in the future.